server/__init__.py

```diff
@@ -1,136 +1,564 @@
+            curGirderType = 'item'
+        else:
+            curGirderType = 'folder'
+        curBindingSpec = wutils.girderInputSpec(
+            hargs[curName],
+            resourceType=curGirderType,
+            dataType='string', dataFormat='string',
+            token=token)
+    else:
+        # inputs that are not of type image, file, or directory
+        # should be passed inline as string from json.dumps()
+        curBindingSpec['mode'] = 'inline'
+        curBindingSpec['type'] = slicerToGirderTypeMap[curType]
+        curBindingSpec['format'] = 'json'
+        curBindingSpec['data'] = hargs['params'][curName]
+
+    return curBindingSpec
+
+
+def createInputTaskSpecFromXML(xmlelt):
+    curName = xmlelt.findtext('name')
+    curType = xmlelt.tag
+
+    curTaskSpec = dict()
+    curTaskSpec['id'] = curName
+    curTaskSpec['type'] = slicerToGirderTypeMap[curType]
+    curTaskSpec['format'] = slicerToGirderTypeMap[curType]
+
+    if curType in ['image', 'file', 'directory']:
+        curTaskSpec['target'] = 'filepath'  # check
+
+    return curTaskSpec
+
+
+def createParamTaskSpecFromXML(xmlelt):
+    curName = xmlelt.findtext('name')
+    curType = xmlelt.tag
+
+    curTaskSpec = dict()
+    curTaskSpec['id'] = curName
+    curTaskSpec['type'] = slicerToGirderTypeMap[curType]
+    curTaskSpec['format'] = slicerToGirderTypeMap[curType]
+
+    defaultValSpec = dict()
+    defaultValSpec['format'] = curTaskSpec['format']
+    strDefaultVal = xmlelt.findtext('default')
+    if strDefaultVal is not None:
+        defaultVal = getDefaultValFromString(strDefaultVal, curType)
+    elif curType == 'boolean':
+        defaultVal = False
+    else:
+        raise Exception(
+            'optional parameters of type %s must '
+            'provide a default value in the xml' % curType)
+    defaultValSpec['data'] = defaultVal
+    curTaskSpec['default'] = defaultValSpec
+
+    return curTaskSpec
+
+
+def createOutputTaskSpecFromXML(xmlelt):
+    curName = xmlelt.findtext('name')
+    curType = xmlelt.tag
+
+    # task spec for the current output
+    curTaskSpec = dict()
+    curTaskSpec['id'] = curName
+    curTaskSpec['type'] = slicerToGirderTypeMap[curType]
+    curTaskSpec['format'] = slicerToGirderTypeMap[curType]
+
+    if curType in ['image', 'file', 'directory']:
+        curTaskSpec['target'] = 'filepath'  # check
+
+    return curTaskSpec
+
+
+def genHandlerToRunCLI(restResource, xmlFile, scriptFile):
+    """Generates a handler to run CLI using girder_worker
+
+    Parameters
+    ----------
+    restResource : girder.api.rest.Resource
+        The object of a class derived from girder.api.rest.Resource to which
+        this handler will be attached
+    xmlFile : str
+        Full path to xml file of the CLI
+    scriptFile : str
+        Full path to .py file containing the code of the clu
+
+    Returns
+    -------
+    function
+        Returns a function that runs the CLI using girder_worker
+    """
```

Comment:
We should use sphinx docstrings like we do in the rest of girder. So this would be:

```
    """Generates a handler to run CLI using girder_worker

    :param restResource: The object of a class derived from girder.api.rest.Resource to which
        this handler will be attached
    :type restResource: `girder.api.rest.Resource`     
    :param xmlFile: Full path to xml file of the CLI
    :type xmlFile: str
    :param scriptFile: Full path to .py file containing the code of the clu
    :type scriptFile: str
    :returns: A function that runs the CLI using girder_worker
    :rtype: callable
    """